frontend/src/components/VHeader/VHeader.vue

```diff
@@ -0,0 +1,51 @@
+<script setup lang="ts">
+import { useElementSize } from "@vueuse/core"
+import { computed, ref, watch } from "vue"
+
+import { useUiStore } from "~/stores/ui"
+
+import VBanners from "~/components/VBanner/VBanners.vue"
+import VHeaderInternal from "~/components/VHeader/VHeaderInternal.vue"
+import VHeaderDesktop from "~/components/VHeader/VHeaderDesktop.vue"
+import VHeaderMobile from "~/components/VHeader/VHeaderMobile/VHeaderMobile.vue"
+
+const props = withDefaults(
+  defineProps<{
+    showBottomBorder?: boolean
+    kind: "internal" | "search-desktop" | "search-mobile"
+    color?: "default" | "complementary"
+  }>(),
+  {
+    showBottomBorder: false,
+    color: "default",
+  }
+)
+
+const headerRef = ref<HTMLElement | null>(null)
+const uiStore = useUiStore()
+const { height } = useElementSize(headerRef)
+watch(height, (height) => {
+  uiStore.setHeaderHeight(height)
+})
+
+const comp = computed(() => {
```

Comment:
```suggestion
const comp = computed(() => {
```

Nit: comp is kinda overloaded with "computed" and "component": How about "headerComponent" instead?